internal/pubsub/acks_test.go

```diff
@@ -266,3 +263,26 @@ func TestConcurrentDoubleAckCausesPanic(t *testing.T) {
 		t.Errorf("panics = %d, want %d", panics, 1)
 	}
 }
+
+func TestSubShutdownCanBeCanceledEvenWithHangingSendAcks(t *testing.T) {
+	ctx := context.Background()
+	m := &driver.Message{AckID: 0} // the batcher doesn't like nil interfaces
+	ds := &ackingDriverSub{
+		q: []*driver.Message{m},
+		sendAcks: func(ctx context.Context, ackIDs []driver.AckID) error {
+			<-ctx.Done()
+			return ctx.Err()
+		},
+	}
+	sub := pubsub.NewSubscription(ds)
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+		sub.Shutdown(ctx)
```

Comment:
This should cancel the sendAcks call.

Reply:
It does. `Subscription.Shutdown` has a call to `s.cancel` that runs when the passed in `ctx` is done. That gets triggered when the 1 msec timeout happens. `s.cancel` is the `cancel` func for the context passed into all the calls to `SendAcks`.